client/apps/settings/client/tags.coffee

```diff
@@ -0,0 +1,107 @@
+Backbone = require 'backbone'
+_ = require 'underscore'
+React = require 'react'
+ReactDOM = require 'react-dom'
+sd = require('sharify').data
+request = require 'superagent'
+Tags = require '../../../collections/tags.coffee'
+Tag = require '../../../models/tag.coffee'
+FilterSearch = React.createFactory require '../../../components/filter_search/index.coffee'
+AddTag = React.createFactory require './add_tag.coffee'
+query = require './tagsQuery.coffee'
+{ div, nav, a, h1, input, button } = React.DOM
+
+module.exports.TagsView = TagsView = React.createClass
+  displayName: 'TagsView'
+
+  getInitialState: ->
+    tags: @props.tags or []
+    public: @props.public or true
+    loading: false
+    errorMessage: ''
+
+  addTag: (name) ->
+    data =
+      name: name
+      public: @state.public
+    new Tag().save data,
+      success: (model, res) =>
+        @setState tags: @state.tags.concat res
+      error: (model, res) =>
+        msg = res?.responseJSON?.message or 'There has been an error. Please contact support.'
+        @flashError msg
+
+  deleteTag: (tag) ->
+    new Tag(id: tag.id).destroy
+      success: (model, res) =>
+        @setState tags: _.filter @state.tags, (t) -> tag.id isnt t.id
+      error: (model, res) =>
+        msg = res?.responseJSON?.message or 'There has been an error. Please contact support.'
+        @flashError msg
+
+  flashError: (msg) ->
+    @setState errorMessage: msg
+    setTimeout ( => @setState(errorMessage: '')), 1000
+
+  searchResults: (results) ->
+    @setState tags: results
+
+  setView: (isPublic) ->
+    tagQuery = query "public: #{isPublic}"
+    request
+      .post sd.API_URL + '/graphql'
+      .set 'X-Access-Token', sd.USER?.access_token
+      .send query: tagQuery
+      .end (err, res) =>
+        return if err or not res.body?.data
+        @setState
+          tags: res.body.data.tags
+          public: isPublic
+
+  getActiveState: (type) ->
+    if @state.public is type then 'is-active' else ''
+
+  render: ->
+    div {
+      className: 'tags-container'
+      'data-loading': @state.loading
+      ref: 'tagsContainer'
```

Comment:
Looks like this is only used for testing, you can locate using the classname too.